returns/interfaces/failable.py

```diff
@@ -0,0 +1,230 @@
+    """
+    Failable laws.
+
+    We need to be sure that ``.rescue`` won't rescue success types.
+    """
+
+    @law_definition
+    def rescue_short_circuit_law(
+        raw_value: _FirstType,
+        container: 'FailableN[_FirstType, _SecondType, _ThirdType]',
+        function: Callable[
+            [_SecondType],
+            KindN['FailableN', _FirstType, _NewFirstType, _ThirdType],
+        ],
+    ) -> None:
+        """Ensures that you cannot rescue a success."""
+        assert_equal(
+            container.from_value(raw_value),
+            container.from_value(raw_value).rescue(function),
+        )
+
+
+class FailableN(
+    ContainerN[_FirstType, _SecondType, _ThirdType],
+    RescuableN[_FirstType, _SecondType, _ThirdType],
+    Lawful['FailableN[_FirstType, _SecondType, _ThirdType]'],
+):
+    """
+    Base type for types that can fail.
+
+    It is a raw type and should not be used directly.
+    Use ``SingleFailableN`` and ``DiverseFailableN`` instead.
+    """
+
+    _laws: ClassVar[Sequence[Law]] = (
+        Law3(_FailableLawSpec.rescue_short_circuit_law),
+    )
+
+
+#: Type alias for kinds with two type arguments.
+Failable2 = FailableN[_FirstType, _SecondType, NoReturn]
+
+#: Type alias for kinds with three type arguments.
+Failable3 = FailableN[_FirstType, _SecondType, _ThirdType]
+
+
+@final
+class _SingleFailableLawSpec(LawSpecDef):
+    """
+    Single Failable laws.
+
+    We need to be sure that ``.map`` and ``.bind``
+    works correctly for ``empty`` property.
+    """
+
+    @law_definition
+    def map_short_circuit_law(
+        container: 'SingleFailableN[_FirstType, _SecondType, _ThirdType]',
+        function: Callable[[_FirstType], _NewFirstType],
+    ) -> None:
+        """Ensures that you cannot map from the `empty` property."""
+        assert_equal(
+            container.empty,
+            container.empty.map(function),
+        )
+
+    @law_definition
+    def bind_short_circuit_law(
+        container: 'SingleFailableN[_FirstType, _SecondType, _ThirdType]',
+        function: Callable[
+            [_FirstType],
+            KindN['SingleFailableN', _NewFirstType, _SecondType, _ThirdType],
+        ],
+    ) -> None:
+        """Ensures that you cannot bind from the `empty` property."""
+        assert_equal(
+            container.empty,
+            container.empty.bind(function),
+        )
+
+
+class SingleFailableN(
+    FailableN[_FirstType, _SecondType, _ThirdType],
+    Lawful['SingleFailableN[_FirstType, _SecondType, _ThirdType]'],
+):
+    """
+    Base type for types that have just only one failed value.
+
+    Like ``Maybe`` types where the only failed value is ``Nothing``.
+    """
+
+    _laws: ClassVar[Sequence[Law]] = (
+        Law2(_SingleFailableLawSpec.map_short_circuit_law),
+        Law2(_SingleFailableLawSpec.bind_short_circuit_law),
+    )
+
+    #: Alias for failed value
+    empty: ClassVar['SingleFailableN[_FirstType, _SecondType, _ThirdType]']
```

Comment:
Let's make it an abstract property, like here: https://github.com/dry-python/returns/blob/master/returns/interfaces/specific/reader.py#L95

So, you will be forced to override it inside yout subtypes.